### PR TITLE
Jest setup

### PR DESCRIPTION
**Pull request by BCerki**
This PR addresses DM-911 installs and configures Jest for front-end testing. It also includes an example unit test suite for the ProgramType component (two tests should pass, one should fail).

Note: I had to use older versions of jest and jest types because the project's typescript version was old. Once Jon finishes the updates, we might be able to go up.

### CODE REVIEW

File: tests/front-end/unit/ProgramType.test.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import {render} from '@testing-library/react'
+// "front-end" means the path "front-end/typescript" per the front-end tsconfig, so have to use the same alias here. IDE might show a TS error but test should run.
+import ProgramType from '../../src/front-end/lib/views/program-type'
+
+
+jest.mock('front-end/lib/framework')
+jest.mock('front-end/lib/views/icon', () => () => (<div>Mocked Icon</div>))
```
Comment (BCerki) [inline tests/front-end/unit/ProgramType.test.tsx:8]: Not confident I've done this correctly

File: tests/front-end/unit/ProgramType.test.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            <ProgramType type_='swu'  />
+          );
+          expect(getByText("Sprint With Us")).toBeTruthy();
+      });
+      it("has correct text for New Product opportunity", () => {
+        const { getByText} = render(
+            <ProgramType type_='new'  />
+          );
+          expect(getByText("New Product")).toBeTruthy();
+      });
```
Comment (BCerki) [inline tests/front-end/unit/ProgramType.test.tsx:29]: This one should fail

**Comment by BryanGK** [comment]
Our team should get an agreement going on when a commit of the `package-lock.json` is appropriate (this might be one). 

You had mentioned having to play with older versions of Jest due to our old versions of everything. Should we wait to commit this until we get our packages up to date?

**Comment by SteveChapmanBCDX** [comment]
Reviewed with team - this issue has been fixed and is included in PR #214.

**Comment by SteveChapmanBCDX** [comment]
Closing ticket.

**Pull request closed without merging**